### PR TITLE
Matriarch Drone Port

### PR DESCRIPTION
**Pull request by NylitisLazarus**
## About The Pull Request

Ports Matriarch drones and associated matrix systems for maint drones from Aurora.

## Why It's Good For The Game

Maint drones could use a little love, and something a bit more interesting to mess with.

## Changelog
:cl:
add: Matriarch drones
add: Matriarch drone fabricator
add: Drone matrix system
add: Matriarch law set
tweak: Changed one drone spawner on Atlas to the new Matriarch type
tweak: Gave maint drones inflatables, RPDs, and engineering tape
code: Added LAZYISIN for quick-checking lists
code: Added isDrone and isMatriarchDrone to helpers
/:cl:

### CODE REVIEW

File: code/__DEFINES/_lists.dm [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*
  * Holds procs to help with list operations
  */
-
+#define LAZYISIN(L, I) (L ? (I in L) : FALSE)
```
Comment (TheLordME) [inline code/__DEFINES/_lists.dm:4]: why not use LAZYACCESS()? the sole difference is that it returns null incase of failure, and not FALSE which both evaluate as False in checks

File: code/modules/mob/living/silicon/robot/drone/matriarchmaintdrone.dm [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/* /datum/ghostspawner/matriarchmaintdrone
```
Comment (TheLordME) [inline code/modules/mob/living/silicon/robot/drone/matriarchmaintdrone.dm:1]: Why are we adding a completely commented out file?

File: code/modules/mob/living/silicon/robot/robot_modules/station/drone.dm [hunk 3]
```diff
@@ -25,6 +25,9 @@
 	src.modules += new /obj/item/floor_painter(src)
 	src.modules += new /obj/item/t_scanner(src)
 	src.modules += new /obj/item/analyzer(src)
+	src.modules += new /obj/item/inflatable_dispenser/robot(src)
```
Comment (TheLordME) [inline code/modules/mob/living/silicon/robot/robot_modules/station/drone.dm:28]: We are not making maintdrones basicly equivalent to engineering borgs

**Review by TheLordME — CHANGES_REQUESTED** [review]
Besides the two minor things I commented on it looks alright.
the module adds you gonna have to revert for approval from me.
However, I dont see any use in this whole thing for us. Maintdrones are an auxilery ghost role that's specifically prohibited from crew interaction.

**Review by silicons — DISMISSED** [review]

File: code/__DEFINES/_lists.dm (hunk 1 above)
Comment (NylitisLazarus) [reply code/__DEFINES/_lists.dm:4]: Cuz I had no idea what I was doing, and forgot about it. Now switched.

File: code/modules/mob/living/silicon/robot/drone/matriarchmaintdrone.dm (hunk 2 above)
Comment (NylitisLazarus) [reply code/modules/mob/living/silicon/robot/drone/matriarchmaintdrone.dm:1]: Also forgotten about. Deleted.

File: code/__DEFINES/spans.dm [hunk 4]
```diff
@@ -125,6 +125,9 @@
 #define SPAN_YELL(str) ("<span class='yell'>" + str + "</span>")
 #define SPAN_YELLOWTEAMRADIO(str) ("<span class='yellowteamradio'>" + str + "</span>")
 
+#define MATRIX_DANGER(X) (SPAN_DANGER(X))
+#define MATRIX_NOTICE(X) (SPAN_NOTICE(X))
```
Comment (Zandario) [inline code/__DEFINES/spans.dm:129]: Why.

**Review by Zandario — CHANGES_REQUESTED** [review]

File: code/__DEFINES/spans.dm (hunk 4 above)
Comment (NylitisLazarus) [reply code/__DEFINES/spans.dm:129]: Good question. Was from the direct port. Now that I get how those work, I can get rid of them for the normal ones.

Comment (Zandario) [reply code/__DEFINES/spans.dm:129]: Please use the normal ones for now yea.

Comment (NylitisLazarus) [reply code/__DEFINES/spans.dm]: Aaaand fixed

File: code/modules/mob/living/silicon/robot/robot_modules/station/drone.dm (hunk 3 above)
Comment (NylitisLazarus) [reply code/modules/mob/living/silicon/robot/robot_modules/station/drone.dm:28]: Per discussion on discord, leaving in the new modules

**Review by Zandario — APPROVED** [review]

**Review by TheLordME — APPROVED** [review]

**Pull request merged**